contracts/source/UniswapOracle.sol

```diff
@@ -66,16 +66,14 @@ contract UniswapOracle {
 			uint112 reserve0;
 			uint112 reserve1;
 			uint256 reserveTimestamp;
-			( , blockNumber, historicPriceCumulativeLast, reserve0, reserve1, reserveTimestamp) = verifyBlockAndExtractReserveData(uniswapV2Pair, minBlocksBack, maxBlocksBack, denominationTokenIs0 ? token0Slot : token1Slot, proofData);
+			( , blockNumber, historicPriceCumulativeLast, reserve0, reserve1, reserveTimestamp) = verifyBlockAndExtractReserveData(uniswapV2Pair, minBlocksBack, maxBlocksBack, denominationTokenIs0 ? token1Slot : token0Slot, proofData);
 			uint256 secondsBetweenReserveUpdateAndHistoricBlock = historicBlockTimestamp - reserveTimestamp;
 			// bring old record up-to-date, in case there was no cumulative update in provided historic block itself
-			if (secondsBetweenReserveUpdateAndHistoricBlock > 0) {
-				// TODO: figure out what denominationTokenIs0 means, re: reserve1/reserve0 ratios
-				historicPriceCumulativeLast += uint(UQ112x112
-					.encode(denominationTokenIs0 ? reserve1 : reserve0)
-					.uqdiv(denominationTokenIs0 ? reserve0 : reserve1)
-					) * secondsBetweenReserveUpdateAndHistoricBlock;
-			}
+			// TODO: figure out what denominationTokenIs0 means, re: reserve1/reserve0 ratios
```

Comment:
As i mentioned in Discord, this one can be left in, because it doesn't change the estimate. the historic block will either always, or never, need to be "trued up"

Reply:
But leaving it out is fine too. saves 200 gas, costs maybe 15 to check?

Reply:
Added back.